server/sonar-server/src/main/java/org/sonar/server/qualityprofile/ws/QProfileSearchAction.java

```diff
@@ -0,0 +1,171 @@
+/*
+ * SonarQube, open source software quality management tool.
+ * Copyright (C) 2008-2014 SonarSource
+ * mailto:contact AT sonarsource DOT com
+ *
+ * SonarQube is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU Lesser General Public
+ * License as published by the Free Software Foundation; either
+ * version 3 of the License, or (at your option) any later version.
+ *
+ * SonarQube is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with this program; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
+ */
+package org.sonar.server.qualityprofile.ws;
+
+import com.google.common.base.Function;
+import com.google.common.collect.Collections2;
+import com.google.common.collect.ImmutableSet;
+import com.google.common.collect.Maps;
+import org.sonar.api.resources.Language;
+import org.sonar.api.resources.Languages;
+import org.sonar.api.server.ws.*;
+import org.sonar.api.server.ws.WebService.NewAction;
+import org.sonar.api.utils.text.JsonWriter;
+import org.sonar.core.qualityprofile.db.QualityProfileDto;
+import org.sonar.server.qualityprofile.QProfile;
+import org.sonar.server.qualityprofile.QProfileLookup;
+import org.sonar.server.qualityprofile.QProfileService;
+
+import java.util.*;
+
+public class QProfileSearchAction implements RequestHandler {
+
+  private static final String FIELD_KEY = "key";
+  private static final String FIELD_NAME = "name";
+  private static final String FIELD_LANGUAGE = "language";
+  private static final String FIELD_IS_INHERITED = "isInherited";
+  private static final String FIELD_IS_DEFAULT = "isDefault";
+  private static final String FIELD_PARENT_KEY = "parentKey";
+  private static final Set<String> ALL_FIELDS = ImmutableSet.of(FIELD_KEY, FIELD_NAME, FIELD_LANGUAGE, FIELD_IS_INHERITED, FIELD_PARENT_KEY, FIELD_IS_DEFAULT);
+
+  private static final String PARAM_LANGUAGE = FIELD_LANGUAGE;
+  private static final String PARAM_FIELDS = "f";
+
+
+  private final Languages languages;
+
+  private final QProfileLookup profileLookup;
+
+  private final QProfileService profileService;
+
+  public QProfileSearchAction(Languages languages, QProfileLookup profileLookup, QProfileService profileService) {
```

Comment:
I would replace QProfileService by QProfileFactory in order to be able to replace mocks in test by DbTester